agentMET4FOF/metrological_agents.py

```diff
@@ -0,0 +1,151 @@
+from typing import Dict, Union
+
+import plotly.graph_objs as go
+from time_series_buffer import TimeSeriesBuffer
+from time_series_metadata.scheme import MetaData
+
+from agentMET4FOF.agents import AgentMET4FOF
+
+
+class MetrologicalAgent(AgentMET4FOF):
+
+    # dict like {
+    #     <from>: {
+    #         "buffer": TimeSeriesBuffer(maxlen=buffer_size),
+    #         "metadata": MetaData(**kwargs).metadata,
+    #     }
+    _input_data: Dict[str, Dict[str, Union[TimeSeriesBuffer, Dict]]]
+    _input_data_maxlen: int
+
+    # dict like {
+    #     <channel> : {
+    #         "buffer" : TimeSeriesBuffer(maxlen=buffer_size),
+    #         "metadata" : MetaData(**kwargs)
+    #     }
+    _output_data: Dict[str, Dict[str, Union[TimeSeriesBuffer, MetaData]]]
+    _output_data_maxlen: int
+
+    def init_parameters(self, input_data_maxlen=25, output_data_maxlen=25):
+        self._input_data = {}
+        self._input_data_maxlen = input_data_maxlen
+        self._output_data = {}
+        self._output_data_maxlen = output_data_maxlen
+
+    def on_received_message(self, message):
+        channel = message["channel"]
+        sender = message["from"]
+
+        if channel == "default":
+            data = message["data"]
+            metadata = None
+            if "metadata" in message.keys():
+                metadata = message["metadata"]
+
+            self._set_input_data(sender, data, metadata)
+
+    def _set_input_data(self, sender, data=None, metadata=None):
+        # create storage for new senders
+        if sender not in self._input_data.keys():
+            self._input_data[sender] = {
+                "metadata": metadata,
+                "buffer": TimeSeriesBuffer(maxlen=self._input_data_maxlen),
+            }
+
+        if not metadata is None:
+            # update received metadata
+            self._input_data[sender]["metadata"] = metadata
+
+        if not data is None:
+            # append received data
+            self._input_data[sender]["buffer"].add(data=data)
+
+    def set_output_data(self, channel, data=None, metadata=None):
+        # create storage for new senders
+        if channel not in self._output_data.keys():
+            self._output_data[channel] = {
+                "metadata": metadata,
+                "buffer": TimeSeriesBuffer(maxlen=self._output_data_maxlen),
+            }
+
+        if not metadata is None:
+            # update received metadata
+            self._output_data[channel]["metadata"] = metadata
+
+        if not data is None:
+            # append received data
+            self._output_data[channel]["buffer"].add(data=data)
```

Comment:
The comments need to be updated to match `_output_data`, instead of `_input_data`. (I created `set_output_data` from `_set_input_data` by copying.)

Reply:
That should be good now. Thanks for the heads up!